src/aavegotchi/realm/parcel-access-right-set-handler.ts

```diff
@@ -0,0 +1,73 @@
+import { log, BigInt } from "@graphprotocol/graph-ts";
+import { ParcelAccessRightSet } from "../../../generated/Realm/RealmDiamond";
+import { AavegotchiLand, Nft } from "../../../generated/schema";
+import { generateNftId } from "../../utils/misc";
+import { AccessRight, ActionRight } from "../../utils/types";
+import { endPreviousRental, createDirectRental, updateLandRights } from "../../utils/land-rentals";
+import { AAVEGOTCHI_LAND } from "../../utils/constants";
+
+/**
+ *  event ParcelAccessRightSet(
+ *        uint256 _realmId, 
+ *        uint256 _actionRight, 
+ *        uint256 _accessRight
+ *  );
+ * 
+ *  Action Rights:
+ *     0: Channeling - only owner
+ *     1: Empty Reservoir - only owner
+ *     2: Equip Installations
+ *     3: Equip Tiles
+ *     4: Unequip Installations
+ *     5: Unequip Tiles
+ *     6: Upgrade Installations
+ *
+ *  Access Rights:
+ *     0: Only Owner
+ *     1: Owner + Lent Out
+ *     2: Whitelisted Only
+ *     3: Allow blacklisted
+ *     4: Anyone
+ */
+export function handleParcelAccessRightSet(event: ParcelAccessRightSet): void {
+  // To start or end a rental, the action right must be Channeling or Empty Reservoir
+  if (event.params._actionRight.gt(BigInt.fromI32(ActionRight.EMPTY_RESERVOIR))) {
+    log.debug("[handleParcelAccessRightSet] Action right {} is not Channeling or Empty Reservoir, tx: {}", [event.params._actionRight.toString(), event.transaction.hash.toHex()]);
+    return;
+  }
+
+  const nftId = generateNftId(AAVEGOTCHI_LAND, event.params._realmId);
+  const nft = Nft.load(nftId);
+
+  if (!nft) {
+    log.debug("[handleParcelAccessRightSet] NFT {} not found, tx: {}", [nftId, event.transaction.hash.toHex()]);
+    return;
+  }
+
+  const land = AavegotchiLand.load(nft.id);
+
+  if (!land) {
+    log.debug("[handleParcelAccessRightSet] Land {} not found, tx: {}", [nft.id, event.transaction.hash.toHex()]);
+    return;
+  }
+
+  // We will always end the previous rental, if it exists
+  endPreviousRental(nft, event.transaction.hash.toHex(), event.block.timestamp);
```

Comment:
Suppose the user emits an event that does not change the rights. Should we end the rental?